src/test/scala/ai/privado/languageEngine/kotlin/tagger/collection/KotlinCollectionTaggerTest.scala

```diff
@@ -0,0 +1,204 @@
+        if (cpg != null) {
+          cpg.close()
+        }
+      }
+    }
+    "tag get Spark Java collection endpoint with method handler as parameter" in {
+      var cpg: Cpg = null
+      try {
+        val fileContents =
+          """
+          |import spark.Spark
+          |
+          |fun main(args: Array<String>) {
+          |    Spark.get("/hello") { req, res ->
+          |        "Hello Spark Kotlin!"
+          |    }
+          |}
+          |""".stripMargin
+        cpg = buildCpg(fileContents)
+        cpg.call.methodFullName(".*.get.*").l.size shouldBe 1
+        cpg.call.head.code.contains("Spark.get") shouldBe true
+
+        val collectionTagger = new MethodFullNameCollectionTagger(cpg, ruleCacheWithCollectionRule(collectionRule))
+        collectionTagger.createAndApply()
+
+        val ingressRules = collectionTagger.getIngressUrls()
+        ingressRules should contain("\"/hello\"")
+
+        val callNode = cpg.call.methodFullName(".*.get.*").head
+        callNode.name shouldBe "get"
+        val tags = callNode.argument.isMethodRef.head.referencedMethod.tag.l
+        tags.size shouldBe 6
+        tags.nameExact(Constants.id).head.value shouldBe ("Collections.Kotlin.HttpFramework")
+        tags.nameExact(Constants.catLevelOne).head.value shouldBe Constants.collections
+        tags.nameExact(Constants.catLevelTwo).head.value shouldBe Constants.default
+        tags.nameExact(Constants.nodeType).head.value shouldBe "REGULAR"
+        tags.nameExact("COLLECTION_METHOD_ENDPOINT").head.value shouldBe "\"/hello\""
+      } finally {
+        if (cpg != null) {
+          cpg.close()
+        }
+      }
+    }
+    "tag get Spark Java collection endpoint with companion method from another class as handler" in {
+      var cpg: Cpg = null
+      try {
+        val fileContents =
+          """
+          |import spark.Request
+          |import spark.Response
+          |import spark.Spark.get
+          |
+          |class HandlerClass {
+          |    companion object {
+          |        val endpointHandler = { req: Request, res: Response ->
+          |            "hello from companion handler"
+          |        }
+          |    }
+          |}
+          |
+          |fun main() {
+          |    get("/hello", HandlerClass.endpointHandler)
+          |}
+          |""".stripMargin
+        cpg = buildCpg(fileContents)
+        cpg.call.methodFullName(".*.get.*").l.size shouldBe 1
+        cpg.call.head.code.contains("endpointHandler") shouldBe true
+
+        val collectionTagger = new MethodFullNameCollectionTagger(cpg, ruleCacheWithCollectionRule(collectionRule))
+        collectionTagger.createAndApply()
+
+        val ingressRules = collectionTagger.getIngressUrls()
+        ingressRules should contain("\"/hello\"")
+
+        val callNode = cpg.call.methodFullName(".*get.*").head
+        callNode.name shouldBe "get"
+        val tags = cpg.method.fullName("<operator>.fieldAccess").head.tag.l
```

Comment:
This is incorrect, the collection point in this case needs to be the method
```
val endpointHandler = { req: Request, res: Response ->
          |            "hello from companion handler"
          |        }

Reply:
This is fixed now.